File: src/components/pages/projects/$projectId/shared/connect-snippets/dotnet/controllers/Controllers/PoliciesController.cs
using Appwrite.Services;
using Microsoft.AspNetCore.Mvc;

namespace ConnectQa.Controllers;

[ApiController]
[Route("v1/policies")]
public class PoliciesController(Project project) : ControllerBase
{
  [HttpPatch]
  public async Task<IActionResult> UpdatePolicy()
  {
    var policy = await project.UpdatePasswordStrengthPolicy(
      min: 8,
      uppercase: true,
      number: true,
      symbols: true
    );

    return Ok(policy.ToMap());
  }

  [HttpGet]
  public async Task<IActionResult> ListPolicies()
  {
    var policies = await project.ListPolicies();

    return Ok(policies.ToMap());
  }
}

File: src/components/pages/projects/$projectId/shared/connect-snippets/ruby/rails/app/controllers/policies_controller.rb
class PoliciesController < ApplicationController
  def update
    policy = project.update_password_strength_policy(
      min: 8,
      uppercase: true,
      number: true,
      symbols: true
    )

    render json: policy.to_map
  end

  def index
    render json: project.list_policies.to_map
  end

  private

  def project
    @project ||= Appwrite::Project.new(APPWRITE_CLIENT)
  end
end
